### PR TITLE
v1.7 fixes, most notably: skins

### PR DESCRIPTION
**Pull request by colinrgodsey**
Requires https://github.com/ProtocolSupport/ProtocolSupport/pull/976

### CODE REVIEW

File: src/protocolsupportpocketstuff/skin/MineskinThread.java [hunk 1]
```diff
@@ -40,6 +41,15 @@ public MineskinThread(BufferedImage skin, boolean isSlim, Consumer<SkinDataWrapp
 		this.skindataUploadedCallback = skindataUploadedCallback;
 	}
 
+	public static BufferedImage resize(BufferedImage inputImage) {
```
Comment (7kasper) [inline src/protocolsupportpocketstuff/skin/MineskinThread.java]: Move this to skin utility class.

File: src/protocolsupportpocketstuff/skin/PeToPcProvider.java [hunk 2]
```diff
@@ -1,46 +1,67 @@
...
 import protocolsupportpocketstuff.util.PacketUtils;
 
+
 public class PeToPcProvider implements PocketPacketListener, Listener {
 
 	private static final ProtocolSupportPocketStuff plugin = ProtocolSupportPocketStuff.getInstance();
 	public static final String TRANSFER_SKIN = "PEApplySkinOnJoin";
 
+	//TODO: remove on player leave
+	private Map<Connection, String> connectionToSkinMap = new WeakHashMap<>();
```
Comment (7kasper) [inline src/protocolsupportpocketstuff/skin/PeToPcProvider.java]: Why use connection map and not the uuid already contained inside the connection metadata?

File: src/protocolsupportpocketstuff/util/PocketInfoReceiver.java [hunk 3]
```diff
@@ -20,10 +20,10 @@ public void onConnect(Connection connection, ClientLoginPacket packet) {
 		// "In general you shouldn't really expect the payload to be sent with psbpe" -Shevchik
 		if (chain != null && clientPayload != null) {
 			clientInfo.put("UUID", UUID.fromString(JsonUtils.getString(chain, "identity")));
-			clientInfo.put("ClientRandomId", clientPayload.get("ClientRandomId").getAsLong());
-			clientInfo.put("DeviceModel", clientPayload.get("DeviceModel").getAsString());
-			clientInfo.put("DeviceOS", clientPayload.get("DeviceOS").getAsInt());
-			clientInfo.put("GameVersion", clientPayload.get("GameVersion").getAsString());
+			if (clientPayload.get("ClientRandomId") != null) clientInfo.put("ClientRandomId", clientPayload.get("ClientRandomId").getAsLong());
```
Comment (7kasper) [inline src/protocolsupportpocketstuff/util/PocketInfoReceiver.java]: 👍

File: src/protocolsupportpocketstuff/skin/PeToPcProvider.java (hunk 2 above)
Comment (colinrgodsey) [reply src/protocolsupportpocketstuff/skin/PeToPcProvider.java]: going to remove this cause it doesnt work anyways- did this before i merged in your last changes. but basically the uuid being stored was the skin uuid, and not the player uuid =\ so needed some association. the code you added looks better tho, so just gonna switch to that

**Comment by 7kasper** [comment]
Yes. Awesome!

**Pull request merged**